### PR TITLE
DYN-6518 - Cherry Picks for 2.19.5

### PR DESCRIPTION
**Pull request by saintentropy**
### Purpose

This Cherry Pick is bring in https://github.com/DynamoDS/Dynamo/pull/14656 and https://github.com/DynamoDS/Dynamo/pull/14744.  These two PR's addressed a regression in behavior for some DropDown nodes in Revit introduced in 2.19.0 and fixed in 3.0.

This original regression was related to this PR https://github.com/DynamoDS/Dynamo/pull/14122

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

Some Drop down nodes had issues when Dynamo is set to Automatic Run Mode in DynamoForRevit.  This change addresses this issue.

### Reviewers

@twastvedt @BogdanZavu 

### FYIs

@Mikhinja @aparajit-pratap @QilongTang

### CODE REVIEW

**Review by twastvedt — APPROVED** [review]

**Review by BogdanZavu — APPROVED** [review]

**Comment by QilongTang** [comment]
The pull request SelfServe results does not seem real, maybe ASM related errors?

File: src/Libraries/CoreNodeModels/DropDown.cs [hunk 1]
```diff
@@ -106,7 +106,6 @@ public int SelectedIndex
                     selectedString = GetSelectedStringFromItem(Items.ElementAt(value));
                 }
 
-                OnNodeModified();
```
Comment (Mikhinja) [inline src/Libraries/CoreNodeModels/DropDown.cs:109]: I believe the fix for one of those two issues in 3.0 caused an issue in Revit ( REVIT-217658 ) where a custom dropdown node of an internal-package for DynamoRevit that needed to update on idle was always one update delayed.

We might want to consider the risks/benefits of this change, and if we will backport this change into 2.19 then in Revit we must also integrate the changes for REVIT-217658. We should also test around nodes that have UI updates to make sure there aren't other issues not yet found.

**Comment by QilongTang** [comment]
@saintentropy @twastvedt Close this PR?

**Comment by twastvedt** [comment]
Closing (won't cherry pick for 2.19) due to the resulting bugs referenced by Vlad above.

**Pull request closed without merging**